File: Cpp/169_Majority_Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int majority = 0;
        for(auto& num : nums){
            if(count == 0) {
                majority = num;
            }
            
            if(num == majority) ++count;
            else --count;
        }
        
        return majority;
    }
};

File: Kotlin/169_Majority_Element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var count = 0
        var majority = 0
        
        nums.forEach{num ->
            if(count == 0) {
                majority = num
            }
            
            if(num == majority) ++count
            else --count
        }
        
        return majority
    }
}